autograder/rest_api/tests/test_views/test_group_views/test_group_views.py

```diff
@@ -606,22 +606,30 @@ def test_bonus_submission_merging(self) -> None:
         self.assertEqual(response.data['bonus_submissions_remaining'], fewer_bonus_submissions)
 
     def test_late_day_merging(self) -> None:
-        self.group1.late_days_used = {
-            self.group1.member_names[0]: 1
-        }
-        self.group1.save()
-        self.group2.late_days_used = {
-            self.group2.member_names[0]: 2
-        }
-        self.group2.save()
+        base_time = timezone.now()
+        # self.project.validate_and_update(allow_late_days=True, closing_time=base_time)
+        self.project.course.validate_and_update(num_late_days=2)
+        self.project.validate_and_update(closing_time=base_time, allow_late_days=True)
+
+        user1 = self.group1.members.first()
+        user2 = self.group2.members.first()
+
+        # before merge, user in group1 uses 1 late day on project
+        # and user in group2 uses 2 late days on project
+        obj_build.make_submission(group=self.group1,
+                                  timestamp=base_time + datetime.timedelta(minutes=2))
+        obj_build.make_submission(group=self.group2,
+                                  timestamp=base_time + datetime.timedelta(days=1, minutes=2))
 
         self.client.force_authenticate(self.admin)
         response = self.client.post(self.get_merge_url(self.group1, self.group2))
         self.assertEqual(status.HTTP_201_CREATED, response.status_code)
-        self.assertEqual(response.data['late_days_used'], {
-            self.group1.member_names[0]: 1,
-            self.group2.member_names[0]: 2
-        })
+
+        # after merge, both users share submissions so they both have used 2 late days on project
+        user1_late_days = ag_models.LateDaysForUser.get(user1, self.course)
+        user2_late_days = ag_models.LateDaysForUser.get(user2, self.course)
+        self.assertEqual(user1_late_days.late_days_used_per_project[self.project.pk], 2)
+        self.assertEqual(user2_late_days.late_days_used_per_project[self.project.pk], 2)
```

Comment:
@james-perretta 

As is, merging groups adds all submissions from each old group to the new group. This obviously affects late day calculations for members of each group. An edge case is that this could cause users from a group to have negative late days remaining after the merge.

Solutions off the top of my head:
1. KISS: submissions don't transfer when merging. Only new submissions will count. I actually think this is fine under most circumstances, but could be problematic if there are last minute / post-deadline merges.
2. FAFO: let admins deal with it (current state)
3. Do something more complicated:
    - Disallow merge if it would make a member's remaining late days go negative? This would still allow a user to lose late days when merging so long as they don't lose more than they have
    - Make submissions from previous groups not count for members of the group they're merging into? This is basically the same as KISS except that it benefits students if for some reason they don't make any submissions with the newly formed group.